### DISCUSSION TITLE
Can't manage to deploy my nuxt project

### DISCUSSION THREAD

**Discussion by ikki-ai** [nuxt/nuxt#23925] [Questions]
### Environment

------------------------------
- Operating System: Windows_NT
- Node Version:     v18.16.0
- Nuxt Version:     3.8.0
- CLI Version:      3.9.1
- Nitro Version:    2.7.0
- Package Manager:  yarn@1.22.19
- Builder:          -
- User Config:      devtools, app, components, modules, snackbar, gtag, pwa, ui, sitemap, build, runtimeConfig, colorMode, cookieControl, postcss
- Runtime Modules:  @nuxt/devtools@1.0.0, nuxt-icon@0.5.0, @nuxt/ui@2.9.0, @vueuse/nuxt@10.5.0, @vueuse/motion/nuxt@2.0.0, @dargmuesli/nuxt-cookie-control@7.0.1, @nuxtjs/device@3.1.1, @nuxt/image@1.0.0, @pinia/nuxt@0.5.1, @pinia-plugin-persistedstate/nuxt@1.1.2, @vite-pwa/nuxt@0.1.1, nuxt-particles@0.1.5, vue3-carousel-nuxt@1.1.0, nuxt-snackbar@1.0.4, nuxt-simple-sitemap@3.4.1, nuxt-gtag@1.1.1
- Build Modules:    -
------------------------------

### Reproduction

...

### Describe the bug

I try to deploy on Digital Ocean One click and on Vercel but i can't manage to understand what's wrong. 
I follow the doc for DO (https://nitro.unjs.io/deploy/providers/digitalocean)
Here is my package : 
 "scripts": {
    "build": "nuxt build",
    "dev": "nuxt dev",
    "generate": "nuxt generate",
    "preview": "nuxt preview",
    "postinstall": "nuxt prepare",
    "start": "node .output/server/index.mjs",
    "lint": "eslint --ext .js,.vue --ignore-path .gitignore ."
  },
  "engines": {
    "node": "^18.16.0"
  },
  "devDependencies": {
    "@dargmuesli/nuxt-cookie-control": "^7.0.1",
    "@nuxt/devtools": "1.0.0",
    "@nuxt/image": "^1.0.0",
    "@nuxt/ui": "^2.9.0",
    "@nuxtjs/device": "^3.1.1",
    "@nuxtjs/eslint-config-typescript": "^12.1.0",
    "@nuxtjs/eslint-module": "^4.1.0",
    "@pinia-plugin-persistedstate/nuxt": "^1.1.2",
    "@pinia/nuxt": "^0.5.1",
    "@popperjs/core": "^2.11.8",
    "@types/js-cookie": "^3.0.5",
    "@types/vue-tel-input": "^2.1.5",
    "@typescript-eslint/eslint-plugin": "^6.9.0",
    "@typescript-eslint/parser": "^6.9.0",
    "@vite-pwa/nuxt": "^0.1.1",
    "@vueuse/core": "^10.5.0",
    "@vueuse/motion": "^2.0.0",
    "@vueuse/nuxt": "^10.5.0",
    "add-to-calendar-button": "^2.4.3",
    "dayjs": "^1.11.10",
    "dayjs-business-time": "^1.0.4",
    "eslint": "^8.52.0",
    "eslint-config-prettier": "^9.0.0",
    "eslint-plugin-import": "^2.29.0",
    "eslint-plugin-jsonc": "^2.10.0",
    "eslint-plugin-prettier": "^5.0.1",
    "eslint-plugin-vue": "^9.18.0",
    "firebase": "^10.5.0",
    "form-data": "^4.0.0",
    "headroom.js": "^0.12.0",
    "js-cookie": "^3.0.5",
    "mitt": "^3.0.1",
    "nodemon": "^3.0.1",
    "nuxt": "^3.8.0",
    "nuxt-gtag": "^1.1.1",
    "nuxt-icon": "^0.5.0",
    "nuxt-particles": "^0.1.5",
    "nuxt-simple-sitemap": "^3.4.0",
    "nuxt-snackbar": "^1.0.4",
    "pinia": "^2.1.7",
    "prettier": "^3.0.3",
    "sass": "^1.69.4",
    "socket.io-client": "^4.7.2",
    "typescript": "^5.2.2",
    "uuid": "^9.0.1",
    "v-calendar": "^3.1.2",
    "vue": "^3.3.6",
    "vue-router": "^4.2.5",
    "vue-signature-pad": "^3.0.2",
    "vue-tel-input": "^8.1.4",
    "vue-tsc": "^1.8.20",
    "vue3-carousel-nuxt": "^1.1.0"
  }
}
And the error i get from DO (the same for Vercel)
I'm sorry to ask here but i really don't get what i'm missing and hope someone could help me :)
Thanks a lot in advance

[2023-10-25 02:47:34] │ [5/5] Building fresh packages...
[2023-10-25 02:47:35] │ $ nuxt prepare
[2023-10-25 02:47:35] │ /bin/sh: 1: nuxt: not found
[2023-10-25 02:47:35] │ error Command failed with exit code 127.
[2023-10-25 02:47:35] │ info Visit https://yarnpkg.com/en/docs/cli/install for documentation about this command.
[2023-10-25 02:47:35] │ 
[2023-10-25 02:47:35] │ unable to invoke layer creator
[2023-10-25 02:47:35] │ installing node_modules: exit status 127
[2023-10-25 02:47:35] │ ERROR: failed to build: exit status 1

```[tasklist]
### Tasks
```

**Comment by ikki-ai** [comment]
I don't know if it helps but when i try to access to the exemple starter with nuxt ui
https://codesandbox.io/p/sandbox/github/nuxt/starter/tree/ui 
there is a similar error

**Comment by ikki-ai** [comment]
And i also try to deploy the Starter with Nuxt UI with the same result.

**Comment by okzmo** [comment]
Hey! I may be mistaken because I'm still quite new when it comes to deploying things, but it seems that all your dependencies are listed under "devDependencies" instead of "dependencies." Generally, "devDependencies" are only used for development-specific dependencies such as testing tools I think.

Again not sure if it's the actual problem but I hope this help!

**Reply by AloisSeckar** [reply]
This shouldn't be the case in Nuxt 3, according to [this post](https://github.com/nuxt/nuxt/discussions/10400#discussioncomment-2615430)

**Reply by okzmo** [reply]
I wasn't aware of that thanks a lot for the insight

**Comment by Norbz** [comment]
Hi,

I had no problem until today and now facing the same issue.
Didn't update the nuxt app or anything, just tried to push a quick CSS fix and my CI is now refusing to build. Exact same error.

I've disable telemetry (which caused me some issue on another project), and even pinned an old docker image that was working previously but without any success...

Nuxt is 3.7.3
Node v20.9.0 (working before)
Npm v10.1.0 (working before)

Quick update :
- Have another project with the exact same CI, using same nuxt version, using same docker image to build the project (node:lts), works perfectly
- Replicated my CI environnment locally with docker, works perfectly
- Yet, that particular project doesn't want to build anymore, even older builds that went well doesn't build anymore
- Tried with nuxt 3.8.0, same thing
- Tried with node 18 and npm 9.8, same thing
- Tried with yarn install instead of npm, same thing.

Edit 2 :
So I don't know why but somehow that particular build was ran with NODE_ENV set to production. This might be a change from bitbucket since I can't find any trace of that in my pipeline. My other project had the node_modules in cache the npm install command would actually just download the cache.

I've resolved the issue by using npm install --include=dev

Maybe that will help :)
